src/IW/Sync/Search.hs

```diff
@@ -28,70 +35,117 @@ import qualified Data.Text as T
 import qualified Data.Vector as V
 
 
--- | Fetch all repositories built with the Haskell language.
-fetchAllHaskellRepos
-    :: ( MonadIO m
+githubSearch
+    :: forall a env m.
+       ( MonadIO m
        , WithError m
        , WithLog env m
+       , FromJSON a
+       , Typeable a
        )
-    => m [GitHub.Repo]
-fetchAllHaskellRepos = liftGitHubSearchToApp searchHaskellRepos
+    => [Text] -- ^ Paths
+    -> Text   -- ^ Query String
+    -> Day    -- ^ From day
+    -> Day    -- ^ To day
+    -> Int    -- ^ Page
+    -> m [a]
+githubSearch paths queryString from to page = liftIO (executeRequest' $ query paths queries) >>= \case
+    Left err -> throwError $ githubErrToAppErr err
+    Right (SearchResult count vec) -> do
+        log Info $ "Query executed with the following paths and query parameters: " <> show paths <> " " <> show queries
+        log Info $ "Fetched total of " <> show count <> " " <> typeName @a <> "s..."
+        searchLimit <- getSearchRateLimit
```

Comment:
Does querying `searchLimit` decreases the search limit? :slightly_smiling_face: 
Also, it would be nice to perform calls to this function only in `Debug` mode, but this is not currently supported by `co-log` and I see that this is a useful feature...

Reply:
Good question. Queries for the search limit do not affect the search limit, so it's fine to make calls to it as many times as you want.